src/finding/mod.rs

```diff
@@ -267,18 +268,54 @@ impl<'w> FindingBuilder<'w> {
     }
 
     pub(crate) fn build(self, workflow: &'w Workflow) -> Result<Finding<'w>> {
+        let locations = self
+            .locations
+            .iter()
+            .map(|l| l.clone().concretize(workflow))
+            .collect::<Result<Vec<_>>>()?;
+
+        let should_ignore = self.ignored_from_inlined_comment(workflow, &locations, self.ident);
+
         Ok(Finding {
             ident: self.ident,
             desc: self.desc,
             determinations: Determinations {
                 confidence: self.confidence,
                 severity: self.severity,
             },
-            locations: self
-                .locations
-                .into_iter()
-                .map(|l| l.concretize(workflow))
-                .collect::<Result<Vec<_>>>()?,
+            locations,
+            ignored_from_inlined_comment: should_ignore,
         })
     }
+
+    fn ignored_from_inlined_comment(
+        &self,
+        workflow: &Workflow,
+        locations: &[Location],
+        id: &str,
+    ) -> bool {
+        let document_lines = &workflow.contents.lines().collect::<Vec<_>>();
+        let line_ranges = locations
+            .iter()
+            .map(|l| {
+                (
+                    l.concrete.location.start_point.row,
+                    l.concrete.location.end_point.row,
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let inlined_ignore = format!("zizmor: ignore[{}]", id);
+        for (start, end) in line_ranges {
+            for document_line in start..(end + 1) {
+                if let Some(line) = document_lines.get(document_line) {
+                    if line.contains(&inlined_ignore) {
+                        return true;
+                    }
+                }
+            }
+        }
```

Comment:
This approach looks decent, but I *think* I'd prefer if we try and do this with the `yamlpath` APIs. My thinking there is that `yamlpath` will ensure that we're only ever parsing within the comment's own span.

The situations in which this matters are pretty obscure, but it's good to avoid them if we can, e.g.:

```yaml
uses: blah
with:
  # false positive
  foo: "zizmor: ignore[rule]"
  # false positive even if we anchor on #
  bar: "# zizmor: ignore[rule]"
  # two matches, unclear which is right
  baz: "something" # zizmor: ignore[rule] # zizmor: ignore[rule] 
```

`yamlpath` should avoid these cases, or at least give us a clear decision procedure about what to do with each :slightly_smiling_face:

Reply:
I totally agree with you. The best I can think of to improve a bit over those false positives is trying with match with regex instead.

On the other hand, perphaps this can be only a first version for this support (we could tag it as `experimental`). Later on, we can iterate over `yamlpath` and replace the implementation which based on the comment extraction API.

You have the call, if you decide this impl. does not work, I'm happy with a few things I've learned along the way 🙂

Reply:
> On the other hand, perphaps this can be only a first version for this support (we could tag it as `experimental`). Later on, we can iterate over `yamlpath` and replace the implementation which based on the comment extraction API.

That works for me -- I'm okay with a non-`yamlpath` approach as long as we don't box ourselves in here.

In the mean time, I *think* we can reduce the false positives here by matching on `# zizmor: ignore[...]` (note the `#` anchor) and searching from the string's right instead of left (i.e. `rfind` instead of `contains`).

(We could then make sure it's a comment by making sure nothing follows it, but that might be overkill.)